src/main/java/de/uni_mannheim/informatik/dws/wdi/WDI_Project_T2/util/Utils.java

```diff
@@ -30,6 +38,26 @@ private static String removeNonDescriptionWords(String text) {
         return text;
     }
 
+    //whitelisting approach
+    public static String keepDescriptionWords(String carDescriptionText) {
+
+        String[] carDescriptionTextArray = carDescriptionText.split("\\W+");
+
+        WhiteListSingleton st = WhiteListSingleton.getInstance();
+
+        String returnString = "";
+
+        for (String str : carDescriptionTextArray) {
+            if (!st.set.contains(str)) {
+                str = "";
+            }
+
+            returnString = returnString + " " + str;
```

Comment:
This will add multiple spaces in case many words are dismissed.
Better:
```java
if (!st.set.contains(str)) {
    returnString += " " + str;
}
```